app/models/merchant.rb

```diff
@@ -3,4 +3,16 @@ class Merchant < ApplicationRecord
 
   has_many :invoices
   has_many :items
+  has_many :transactions, through: :invoices
+
+  def total_revenue
```

Comment:
#start with invoices and transactions
    #transaction need to have result success to be paid invoice
    #total_revnue = quantity * unit_price (only invoice item table has relationship)